### PR TITLE
Replacing Unnecessary Strings With Characters To Reduce The Load On The Compiler

### PR DESCRIPTION
**Pull request by TaMaN2031A**
### Description
Replacing some unnecessary strings such as "/", "+", and "=" with char representation of these symbols.
Solver this issue #871

### CODE REVIEW

File: moveit_core/robot_state/src/robot_state.cpp [hunk 1]
```diff
@@ -2315,19 +2315,19 @@ void RobotState::printTransforms(std::ostream& out) const
   const std::vector<const JointModel*>& jm = robot_model_->getJointModels();
   for (const JointModel* joint : jm)
   {
-    out << "  " << joint->getName();
+    out << ' ' << ' ' << joint->getName();
```
Comment (sjahr) [inline moveit_core/robot_state/src/robot_state.cpp:2318]: No need to do this twice
```suggestion
    out << '  ' << joint->getName();
```

Comment (sjahr) [reply moveit_core/robot_state/src/robot_state.cpp:2318]: :point_up: This applies to multiple changes in this code

File: moveit_core/robot_state/src/robot_state.cpp [hunk 2]
```diff
@@ -2315,19 +2315,19 @@ void RobotState::printTransforms(std::ostream& out) const
...
     const int idx = joint->getJointIndex();
     if (dirty_joint_transforms_[idx])
       out << " [dirty]";
-    out << ": ";
+    out << ":" << ' ';
```
Comment (sjahr) [inline moveit_core/robot_state/src/robot_state.cpp:2322]: ":" is still string, it should be
```suggestion
    out << ': ';
```

Comment (sjahr) [reply moveit_core/robot_state/src/robot_state.cpp:2322]: :point_up:  This applies to multiple changes in this code

File: moveit_planners/pilz_industrial_motion_planner/include/joint_limits_copy/joint_limits.hpp [hunk 3]
```diff
@@ -99,13 +94,13 @@ struct JointLimits
   {
     std::stringstream ss_output;
 
-    ss_output << "  has position limits: " << (has_position_limits ? "true" : "false") << "[" << min_position << ", "
```
Comment (sjahr) [inline moveit_planners/pilz_industrial_motion_planner/include/joint_limits_copy/joint_limits.hpp:102]: I think you could replace the things like ", " or "]\n" too

File: moveit_planners/pilz_industrial_motion_planner/include/joint_limits_copy/joint_limits.hpp [hunk 4]
```diff
@@ -64,28 +64,23 @@ struct JointLimits
 
     if (has_position_limits)
     {
-      ss_output << "  position limits: "
```
Comment (sjahr) [inline moveit_planners/pilz_industrial_motion_planner/include/joint_limits_copy/joint_limits.hpp:67]: I think you could replace the things like ", " or "]\n" too

**Review by sjahr — COMMENTED** [review]
Thanks for proposing these changes. I have some comments:

File: moveit_core/robot_state/src/robot_state.cpp (hunk 1 above)
Comment (TaMaN2031A) [reply moveit_core/robot_state/src/robot_state.cpp:2318]: Is it possible to put two chars within the same ' '? It prints a numerical value when I do so

Comment (TaMaN2031A) [reply moveit_core/robot_state/src/robot_state.cpp:2318]: @sjahr

**Comment by TaMaN2031A** [comment]
@tylerjw May you review the pr? I don't want to open a new one till this one is closed, thank you

File: moveit_core/collision_detection/src/collision_common.cpp [hunk 5]
```diff
@@ -59,7 +59,7 @@ void CollisionResult::print() const
     for (const auto& contact : contacts)
     {
       RCLCPP_DEBUG_STREAM_THROTTLE(LOGGER, clock, LOG_THROTTLE_PERIOD,
-                                   "\t" << contact.first.first << ", " << contact.first.second);
+                                   "\t" << contact.first.first << ',' << ' ' << contact.first.second);
```
Comment (tylerjw) [inline moveit_core/collision_detection/src/collision_common.cpp:62]: This seems odd to me, it is two characters so a string literal is fine here.

File: moveit_core/robot_model/src/joint_model.cpp [hunk 6]
```diff
@@ -292,22 +292,22 @@ std::ostream& operator<<(std::ostream& out, const VariableBounds& b)
 {
   out << "P." << (b.position_bounded_ ? "bounded" : "unbounded") << " [";
   printBoundHelper(out, b.min_position_);
-  out << ", ";
```
Comment (tylerjw) [inline moveit_core/robot_model/src/joint_model.cpp:295]: Same comment on these

**Review by tylerjw — COMMENTED** [review]
I'm mostly confused by this PR. I'll go read the linked issue to see if I can understand what the point is.

File: moveit_core/collision_detection/src/collision_common.cpp (hunk 5 above)
Comment (TaMaN2031A) [reply moveit_core/collision_detection/src/collision_common.cpp:62]: Thank you for your response <3
I'll modify it and change only the strings which contain one character in it

**Comment by TaMaN2031A** [comment]
Could you review it? @tylerjw

**Comment by ibrahiminfinite** [comment]
@TaMaN2031A could you fix the clang-tidy issues ?

**Comment by tylerjw** [comment]
I'm closing as this seems to be abandoned.

**Pull request closed without merging**